Seat ticket impl

STOzaki opened the pull request:
Almost finished except that I have not personally tested it yet

STOzaki committed:
changed int to Integer to handle nulls

STOzaki committed:
finished seat Ticket controller

This controller holds most of the booking and cancelling implementations

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	private String cancellationAPI;
+	
+	/**
+	 * Helper method to reduce the amount of repetitive code required for "get-all"
```
Since this isn't used only for "get-all" methods in this project, I suggest removing that part of the sentence.

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	 * @return the response the server sent
+	 */
+	private <T> ResponseEntity<T> methodCall(final String url, final HttpMethod method) {
+		return restTemplate.exchange(url, method, null,
```
To make the difference between this and the following method clearer, this method could simply call it with `null` for the final parameter.

STOzaki replied:
However, it would be a little too complicated to fix this in the short time that we have

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	public ResponseEntity<SeatLocation> getAllSeatsWithPlan(
+			@PathVariable final String flightId) {
+		// FIXME: Search service doesn't yet provide this
+		String url = searchAPI + "/seats?flight=" + flightId;
```
This is a short enough line that I would simply inline it into the sole use, but if you prefer to leave it separate that's fine too.

STOzaki replied:
Yeah, that is true.

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	}
+
+	/**
+	 * Get list of seats on a given flight
```
This comment is not correct for this method.

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	public ResponseEntity<Ticket> getTicket(
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId) {
+		String url = bookingAPI + "/details/flights/" + flightId + "/rows/" +
```
You might want to use `String.format` rather than repeated concatenation; it'll be shorter, and would be a little easier to debug in some cases if you forgot something.

STOzaki replied:
But, in the short time that we have, it would be difficult to do it for all of the concatenations.

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final PaymentAmount pay) {
+		// may need to just create a new object
+		if (pay.getPrice() == null) {
```
Personally I would try to reduce the amount of duplication between the two branches of the `if`, but if that's too complicated for you that's fine.

STOzaki replied:
Yes, but it would be a little too complicated

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+			@PathVariable final int flightId, @PathVariable final int row,
+			@PathVariable final String seatId, @RequestBody final User reserver) {
+		// not sure if the request body int worked
+		// it should through an error if they do not give a reserver (right?)
```
You mean "throw" an error; "through" is not a verb. (But are these comments still applicable?)

STOzaki replied:
Yeah, I do not think they are applicable

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+			@PathVariable final String seatId, @RequestBody final User reserver) {
+		// not sure if the request body int worked
+		// it should through an error if they do not give a reserver (right?)
+		HttpEntity<User> body = new HttpEntity<>(reserver);
```
I would inline the `body` variable into its sole use in each of these methods.

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	 * @return list of tickets
+	 */
+	@GetMapping("/booking/{bookingCode}")
+	public ResponseEntity<Ticket> getAllTicketsWithBookingCode(
```
Surely this should be just `getTicketWithBookingCode`? (There should be at most one ...)

kingjon3377 commented on src/main/java/com/st/utopia/agent/controller/SeatTickets.java:
```diff
@@ -0,0 +1,189 @@
+	 * Cancel reservation
+	 */
+	@DeleteMapping("/booking/{bookingCode}")
+	public ResponseEntity<Object> DeleteReservation(
```
A method name should never start with a capital letter in Java.

kingjon3377 reviewed (changes requested):
The one problem I would see as serious enough to block this is the method with a capitalized name; beyond that there are a number of "I would have done that in this other way" comments, but those are mostly not even suggestions that *you* do them that way.

STOzaki committed:
improve comments http method call

STOzaki committed:
improve doc for methods

specifically for getting ticket based on flight, row, and seat

STOzaki committed:
remove old comment on booking a flight

STOzaki committed:
remove intermediary variable

STOzaki committed:
improve naming method for getting ticket via bookingCode

STOzaki committed:
make the DeleteReservation lower-case

java standard convention

STOzaki committed:
remove unnecessary semicolon

in /UtopiaAirlinesAgentAPI/src/main/java/com/st/utopia/agent/controller/SeatTickets.java

STOzaki committed:
update logic for the updateTicket method on SeatTicket controller

STOzaki committed:
update logic for the putBookingCode method on SeatTicket controller
